zerver/webhooks/pivotal/tests.py

```diff
@@ -186,5 +198,41 @@ def test_type_changed(self) -> None:
             "type_changed", expected_topic, expected_message, content_type="application/xml"
         )
 
+    def test_bad_payload(self) -> None:
+        bad = ("foo", None, "bar")
+        with self.assertRaisesRegex(AssertionError, "Unable to handle Pivotal payload"):
+            with mock.patch(
+                "zerver.webhooks.pivotal.view.api_pivotal_webhook_v3", return_value=bad
+            ):
+                self.check_webhook("accepted", expect_topic="foo")
+
+    def test_bad_request(self) -> None:
+        request = mock.MagicMock()
+        hamlet = self.example_user("hamlet")
+        bad = {
+            "project": {
+                "name": "",
+                "id": "",
+            },
+            "primary_resources": [
+                {
+                    "url": "",
+                    "id": 0,
+                    "name": "",
+                }
+            ],
+            "msg": "foo",
+        }
```

Comment:
We've been trying to avoid using mocks at all in the webhook tests in favor of just verifying end-to-end output; can this just be a fixture that's labeled as invalid and then use `self.check_webhook`?

Reply:
DONE, squashed and pushed.